packages/utils/src/facility/utils.ts

```diff
@@ -0,0 +1,183 @@
+import { getEnvVarOrFail } from "@metriport/shared/common/env-var";
+import { MetriportError } from "@metriport/shared";
+import axios from "axios";
+import { access } from "node:fs/promises";
+import { pipeline } from "node:stream/promises";
+import path from "path";
+import { createReadStream, constants as FS } from "node:fs";
+import csvParser from "csv-parser";
+import { Facility as SdkFacility } from "@metriport/api-sdk/medical/models/facility";
+import { sleep } from "@metriport/shared";
+
+interface FacilityWithOid extends SdkFacility {
+  oid: string;
+}
+
+const internalUrl = getEnvVarOrFail("API_URL");
+
+interface NpiRow {
+  npi: string;
+}
+
+//eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function getCwFacility(cxId: string, id: string, oid: string): Promise<any> {
+  const resp = await axios.get(
+    internalUrl + `/internal/commonwell/ops/organization/${oid}?facilityId=${id}&cxId=${cxId}`
+  );
+  if (!resp.data) throw new Error(`CW Organization not returned`);
+  return resp.data;
+}
+
+//eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function getCqFacility(cxId: string, id: string, oid: string): Promise<any> {
+  const resp = await axios.get(
+    internalUrl +
+      `/internal/carequality/ops/directory/organization/${oid}?facilityId=${id}&cxId=${cxId}`
+  );
+  if (!resp.data) throw new Error(`CQ Organization not returned`);
+  return resp.data;
+}
+
+export async function readNpisFromCsv(inputPath: string): Promise<string[]> {
+  const npis: string[] = [];
+
+  const parser = csvParser({
+    headers: ["npi"],
+    skipLines: 1,
+  });
+
+  parser.on("data", (row: NpiRow) => {
+    if (row.npi && row.npi.trim()) {
+      npis.push(row.npi.trim());
+    }
+  });
+
+  const filePath = path.resolve(inputPath);
+
+  try {
+    await access(filePath, FS.R_OK);
+  } catch {
+    throw new MetriportError("File does not exist or is not readable.", undefined, {
+      inputPath: filePath,
+    });
+  }
+
+  await new Promise<void>((resolve, reject) => {
+    parser.once("end", resolve);
+    parser.once("error", reject);
+    pipeline(createReadStream(filePath), parser).catch(reject);
+  });
+
+  return npis;
+}
+
+export async function getCwFacilitySafe(
+  cxId: string,
+  id: string,
+  oid: string
+  //eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Promise<any | null> {
+  try {
+    return await getCwFacility(cxId, id, oid);
+  } catch {
+    return null;
+  }
+}
+
+export async function getCqFacilitySafe(
+  cxId: string,
+  id: string,
+  oid: string
+  //eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Promise<any | null> {
+  try {
+    return await getCqFacility(cxId, id, oid);
+  } catch {
+    return null;
+  }
+}
+
+export async function getFacilityByNpi(cxId: string, npi: string): Promise<FacilityWithOid | null> {
```

Comment:
Can we name this something else? Confusing because we already have a getFacilityByNpi in core that does something else